### DISCUSSION TITLE
sBrick support

### DISCUSSION THREAD

**Discussion by benredstar** [pybricks/support#1606] [Ideas]
Hey there 

Don't suppose it's possible to support sBricks?

So hypothetically, if a LEGO Hub running pyBricks could in turn connect to and send commands to sBricks, activating their motors etc.

Just wondering...

/ben

**Comment by Mantarri** [comment]
I've been working on this type of functionality, but for my BuWizz 2.0 (I don't have an SBrick.) I've had to take a haitus until I finish this semester, but I can share what I've found so far. First of all was finding source code for apps which I knew worked with BuWizz, in this case being BuWizz's own [example app source](https://github.com/BuWizz/BuWizz-Android-Demo), along with the [source of BrickController 2](https://github.com/imurvai/brickcontroller2). I also looked at BuWizz's API reference for good measure - not sure if SBrick has made their API public or not.

Anyway, from there I first worked on just creating a Python script for my PC, using the Bleak python package for connecting to bluetooth devices. From there, I fiddled around until I had a script that could successfully connect to and control my brick (I could throw it up into a gist and link it if that'd be helpful as a reference. It wouldn't be 1:1 for an SBrick, but it shouldn't be all that different.) From there, I've been working on adapting PyBricks code for connecting to an Xbox controller, and LEGO train remote as a reference for how to "translate" my script into a module for the PyBricks firmware. A look at some of my discoveries and questions in this process can be seen in discussion #1452.
